test/Microsoft.Azure.Relay.UnitTests/RunTimeTests.cs

```diff
@@ -153,6 +153,77 @@ async Task ConcurrentClientsTest(EndpointTestType endpointTestType)
             }
         }
 
+        [Theory, DisplayTestMethodName]
+        [MemberData(nameof(AuthenticationTestPermutations))]
+        async Task RequestHeadersTest(EndpointTestType endpointTestType)
+        {
+            HybridConnectionListener listener = null;
+            try
+            {
+                listener = this.GetHybridConnectionListener(endpointTestType);
+                var client = GetHybridConnectionClient(endpointTestType);
+
+                var listenerRequestTcs = new TaskCompletionSource<IDictionary<string, string>>();
+                listener.AcceptHandler = (context) =>
+                {
+                    try
+                    {
+                        var headers = new Dictionary<string, string>();
+                        foreach (string headerName in context.Request.Headers.AllKeys)
+                        {
+                            headers[headerName] = context.Request.Headers[headerName];
+                        }
+
+                        listenerRequestTcs.SetResult(headers);
+                        return Task.FromResult(true);
+                    }
+                    catch (Exception e)
+                    {
+                        listenerRequestTcs.TrySetException(e);
+                        throw;
+                    }
+                };
+
+                TestUtility.Log($"Opening {listener}");
+                await listener.OpenAsync(TimeSpan.FromSeconds(30));
+
+                const string ExpectedRequestHeaderName = "CustomHeader1";
+                const string ExpectedRequestHeaderValue = "Some value here; other-value/here.";
```

Comment:
>Some value here; other-value/here. [](start = 59, length = 34)

Does this include the invalid values?

Reply:
No, this value is valid.

---
In reply to: [266133306](https://github.com/Azure/azure-relay-dotnet/pull/97#discussion_r266133306) [](ancestors = 266133306)

Reply:
I don't know if it's worthwhile to explictly test for invalid values

---
In reply to: [266133760](https://github.com/Azure/azure-relay-dotnet/pull/97#discussion_r266133760) [](ancestors = 266133760,266133306)

Reply:
added in the 3rd iteration.

---
In reply to: [266134567](https://github.com/Azure/azure-relay-dotnet/pull/97#discussion_r266134567) [](ancestors = 266134567,266133760,266133306)